### DISCUSSION TITLE
Unable to deploy sunbird Ed portal version 4.10.0- error ../src/app/client/node_modules/iltorb: Command failed

### DISCUSSION THREAD

**Discussion by paritshivani** [Sunbird-Ed/Community#683] [Installation]
Hello team,
We are trying to deploy Sunbird Ed portal version 4.10.0.
While installing client dependancies, we are getting following issue
`error /var/lib/jenkins/workspace/Build/Core/Player/src/app/client/node_modules/iltorb: Command failed.
14:28:01  Exit code: 1
14:28:01  Command: node ./scripts/install.js || node-gyp rebuild
14:28:01  Arguments: 
14:28:01  Directory: /var/lib/jenkins/workspace/Build/Core/Player/src/app/client/node_modules/iltorb
14:28:01  Output:
14:28:01  info looking for cached prebuild @ /var/lib/jenkins/.npm/_prebuilds/c57c4d-iltorb-v2.4.5-node-v83-linux-x64.tar.gz
14:28:01  http request GET https://github.com/nstepien/iltorb/releases/download/v2.4.5/iltorb-v2.4.5-node-v83-linux-x64.tar.gz
14:28:01  http 404 https://github.com/nstepien/iltorb/releases/download/v2.4.5/iltorb-v2.4.5-node-v83-linux-x64.tar.gz
14:28:01  WARN install No prebuilt binaries found (target=14.19.0 runtime=node arch=x64 libc= platform=linux)
14:28:01  gyp info it worked if it ends with ok
14:28:01  gyp info using node-gyp@5.1.0
14:28:01  gyp info using node@14.19.0 | linux | x64
14:28:01  gyp info find Python using Python version 2.6.9 found at "/usr/local/bin/python"
14:28:01  gyp info spawn /usr/local/bin/python
14:28:01  gyp info spawn args [
14:28:01  gyp info spawn args   '/var/lib/jenkins/.nvm/versions/node/v14.19.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
14:28:01  Could not find platform independent libraries <prefix>
14:28:01  gypCould not find platform dependent libraries <exec_prefix>
14:28:01  Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
14:28:01   info spawn args   'binding.gyp',
14:28:01  gyp info spawn args   '-f',
14:28:01  gyp info spawn args   'make',
14:28:01  gyp info spawn args   '-I',
14:28:01  gyp info spawn args   '/var/lib/jenkins/workspace/Build/Core/Player/src/app/client/node_modules/iltorb/build/config.gypi',
14:28:01  gyp info spawn args   '-I',
14:28:01  gyp info spawn args   '/var/lib/jenkins/.nvm/versions/node/v14.19.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
14:28:01  gyp info spawn args   '-I',
14:28:01  gyp info spawn args   '/var/lib/jenkins/.cache/node-gyp/14.19.0/include/node/common.gypi',
14:28:01  gyp info spawn args   '-Dlibrary=shared_library',
14:28:01  gyp info spawn args   '-Dvisibility=default',
14:28:01  gyp info spawn args   '-Dnode_root_dir=/var/lib/jenkins/.cache/node-gyp/14.19.0',
14:28:01  gyp info spawn args   '-Dnode_gyp_dir=/var/lib/jenkins/.nvm/versions/node/v14.19.0/lib/node_modules/npm/node_modules/node-gyp',
14:28:01  gyp info spawn args   '-Dnode_lib_file=/var/lib/jenkins/.cache/node-gyp/14.19.0/<(target_arch)/node.lib',
14:28:01  gyp info spawn args   '-Dmodule_root_dir=/var/lib/jenkins/workspace/Build/Core/Player/src/app/client/node_modules/iltorb',
14:28:01  gyp info spawn args   '-Dnode_engine=v8',
14:28:01  gyp info spawn args   '--depth=.',
14:28:01  gyp info spawn args'import site' failed; use -v for traceback
14:28:01     '--no-parallel',
14:28:01  gyp info spawn args   '--generator-output',
14:28:01  gyp info spawn args   'build',
14:28:01  gyp info spawn args   '-Goutput_dir=.'
14:28:01  gyp info spawn args ]
14:28:01  Traceback (most recent call last):
14:28:01    File "/var/lib/jenkins/.nvm/versions/node/v14.19.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 7, in <module>
14:28:01      import os
14:28:01  ImportError: No module named os
14:28:01  gyp ERR! configure error 
14:28:01  gyp ERR! stack Error: `gyp` failed with exit code: 1
14:28:01  gyp ERR! stack     at ChildProcess.onCpExit (/var/lib/jenkins/.nvm/versions/node/v14.19.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
14:28:01  gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
14:28:01  gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
14:28:01  gyp ERR! System Linux 5.15.0-1050-azure
14:28:01  gyp ERR! command "/var/lib/jenkins/.nvm/versions/node/v14.19.0/bin/node" "/var/lib/jenkins/.nvm/versions/node/v14.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
14:28:01  gyp ERR! cwd /var/lib/jenkins/workspace/Build/Core/Player/src/app/client/node_modules/iltorb
14:28:01  gyp ERR! node -v v14.19.0
14:28:01  gyp ERR! node-gyp -v v5.1.0
14:28:01  gyp ERR! not ok`

We checked that  **iltorb** is **deprecated** and it is internal dependancy of **gulp-brotli**. 

How can we resolve this deployment issue?

**Comment by gandham-santhosh** [comment]
@rajeevsathish @vinukumar-vs  please check this

**Comment by rajeevsathish** [comment]
@paritshivani is this in local or in the server and what is the node version that you are using?

**Reply by paritshivani** [reply]
@rajeevsathish , Its on server
for sunbird Ed 4.10, we are using node 14.19.0
We checked that **iltorb** is deprecated and it is internal dependancy of **gulp-brotli**.

**Comment by rajeevsathish** [comment]
@paritshivani please start using release-5.X.X and also if any gyp issue is there this is more of a system specific please see how can you add and fix it

**Reply by paritshivani** [reply]
@rajeevsathish , as we have so many customisations in project we cant upgrade to next version at this stage. 
@gandham-santhosh , @sravani-64, Can you please check the system specific issue @rajeevsathish suggesting about?

**Comment by anisha147** [comment]
Hi @paritshivani 

We hope the suggestion shared by @rajeevsathish helped resolve the issue. If so, could you please close this thread from your end?

cc: @rajeevsathish @gandham-santhosh
